lib/charms/mongodb/v0/mongodb_provider.py

```diff
@@ -111,6 +140,22 @@ def _on_relation_event(self, event):
         departed_relation_id = None
         if type(event) is RelationBrokenEvent:
             departed_relation_id = event.relation.id
+            # we receives relation broken events when: the relation has been removed, units are
+            # scaling down, or the application has been removed. Only proceed to process user
+            # removal if the relation has been removed.
+            relation_departed_key = f"relation_{event.relation.id}_departed"
+            if relation_departed_key not in self.charm.app_peer_data:
+                logger.info(
+                    "Deferring, must wait for relation departed hook to decide if relation should be removed."
+                )
+                event.defer()
+                return
+
+            if not json.loads(self.charm.app_peer_data[relation_departed_key]):
```

Comment:
nits:
Maybe we can have a function "is_scalind_down" that will do this check? so the if will look like

if is_scaling_down():
   logger.info(...)
   return

Reply:
Great idea, especially since multiple relation interfaces will implement this. I will do this right now